decidim-proposals/app/commands/decidim/proposals/admin/reset_participatory_text.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Proposals
+    module Admin
+      # A command with all the business logic related with an admin reseting participatory text proposals.
+      class ResetParticipatoryText < Rectify::Command
+        # Public: Initializes the command.
+        #
+        # form - A PreviewParticipatoryTextForm form object with the params.
+        def initialize(component)
+          @component = component
+        end
+
+        # Executes the command. Broadcasts these events:
+        #
+        # - :ok when everything is valid.
+        # - :invalid if the form wasn't valid and we couldn't proceed.
+        #
+        # Returns nothing.
+        def call
+          transaction do
+            reset_drafts
+          end
+
+          broadcast(:ok)
+        end
+
+        private
+
+        attr_reader :form
+
+        def reset_drafts
```

Comment:
```suggestion
        def delete_draft_proposals
```

Reply:
Ok, as we talked, I'll change the reset terminology for discard